### DISCUSSION TITLE
Version number needed to create a nedit-ng OpenBSD port

### DISCUSSION THREAD

**Discussion by cgdae** [eteran/nedit-ng#349] [General]
I have a prototype OpenBSD port for nedit-ng (basically a 25-line Makefile), currently hard-coded to the newest nedit-ng in git, a5be3d1e. It seems to build, install and uninstall ok.

However to add this as an official OpenBSD port i'll need an OpenBSD port version number - numbers separated by dots, with an optional suffix such as `rc2`. See https://man.openbsd.org/packages-specs.7 for details.

I've read elsewhere that the first release of nedit-ng is intended to be version 6.0. So would it make sense to call the OpenBSD port `nedit-ng-5.9`, or `nedit-ng-6.0alpha1`? The only technical requirement is that the version number compares less than 6.0, so that the ordering is ok when the first official nedit-ng release is made.

Thanks for any suggestions here.

**Comment by eteran** [comment]
Ah, good question, there's two versions at play.

1. NEdit-NG typically has a version like `2023.1` meaning the first release of 2023. I am planning on doing a new versioned release soon, if getting it into the OpenBSD repos requires this, I can prioritize it :-).

2. Yes, there is plans to eventually switch to calling this project NEdit-6.0, (note, not "NG") but admittedly, development has slowed down a bit recently. The rebranding is something I'd need to continue to coordinate with the original owners of NEdit, but in my mind it's still "the plan".

So for now at least, I'd stick with the NG versioning if that makes sense to you since the second version at play is a full rebrand in order to "take over" primary development of "classic NEdit"

**Comment by cgdae** [comment]
Thanks, that all makes sense. I'll wait for nedit-ng-2023.1 to be released.

[This is my first OpenBSD port, so it'll probably take a while to get things into a state that is acceptable to the ports people anyway.]

**Comment by cgdae** [comment]
Update: the OpenBSD folk have fixed up my attempt at a port (see https://marc.info/?l=openbsd-ports&m=168846206119856&w=2), so it's ready to go when nedit-ng is next released.

**Comment by eteran** [comment]
@cgdae I totally dropped the ball on this one. Thanks to a long time user, I recently squashed a couple of bugs. Once he confirms the latest fix, I will plan to do a release ASAP.

I am so sorry that it took, basically a whole year for me to follow up!

**Reply by julian-smith-artifex-com** [reply]
That's great, no problem with the delay :-)

**Comment by eteran** [comment]
@cgdae / @julian-smith-artifex-com 

It took WAY too long and I'm again, very sorry that it took so long, but 2025.1 has been officially released!
